File: LeetCode/[M] - 435. Non-overlapping Intervals/solution.go
package main

import "slices"

func eraseOverlapIntervals(intervals [][]int) int {
	slices.SortFunc(intervals, func(a, b []int) int {
		return a[1] - b[1]
	})
	cnt := 1
	for i := 1; i < len(intervals); i++ {
		if intervals[i-1][1] <= intervals[i][0] {
			cnt += 1
		} else {
			intervals[i] = intervals[i-1]
		}
	}
	return len(intervals) - cnt
}

File: LeetCode/[M] - 435. Non-overlapping Intervals/solution.py
from typing import List


class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[1])

        cnt = 1
        for i in range(1, len(intervals)):
            if intervals[i - 1][1] <= intervals[i][0]:
                cnt += 1
            else:
                intervals[i] = intervals[i - 1]
        return len(intervals) - cnt
